src/platform/GenericOTARequestorDriver.h

```diff
@@ -42,23 +43,45 @@ class GenericOTARequestorDriver : public OTARequestorDriver
         mImageProcessor = processor;
     }
 
+    // Set the timeout (in seconds) for the Default Provider timer
+    void SetDefaultProvidersTimeoutSec(uint32_t timeout) { mDefaultProviderTimeoutSec = timeout; }
+
+    //// Virtual methods from OTARequestorDriver
     bool CanConsent() override;
     uint16_t GetMaxDownloadBlockSize() override;
-
     void HandleError(UpdateFailureState state, CHIP_ERROR error) override;
+    void HandleIdleState() override;
     void UpdateAvailable(const UpdateDescription & update, System::Clock::Seconds32 delay) override;
     void UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay) override;
     void UpdateDownloaded() override;
     void UpdateConfirmed(System::Clock::Seconds32 delay) override;
     void UpdateSuspended(System::Clock::Seconds32 delay) override;
     void UpdateDiscontinued() override;
     void UpdateCancelled() override;
+    void ScheduleDelayedAction(System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState) override;
+    void CancelDelayedAction(System::TimerCompleteCallback action, void * aAppState) override;
+    void ProcessAnnounceOTAProviders(const ProviderLocationType & providerLocation,
+                                     app::Clusters::OtaSoftwareUpdateRequestor::OTAAnnouncementReason announcementReason) override;
+    void DriverSendQuery() override;
```

Comment:
nit: In a class where it is already called `GenericOTARequestorDriver`, putting `Driver` in the function name seems redundant. Suggest calling this `SendQuery` or `RequestSendQuery` or `AttemptSendQuery`

Reply:
Let's keep the name for now as it's referenced in our discussions.

Reply:
Renamed